pkg/controller/useraccount/useraccount_controller.go

```diff
@@ -191,29 +211,148 @@ func (r *ReconcileUserAccount) ensureIdentity(logger logr.Logger, userAcc *toolc
 	return identity, false, nil
 }
 
-// updateStatus updates user account status to given status with errMsg but only if the current status doesn't match
-// If the current status already set to desired state then this method does nothing
-func (r *ReconcileUserAccount) updateStatus(userAcc *toolchainv1alpha1.UserAccount, status toolchainv1alpha1.StatusUserAccount, errMsg string) error {
-	if userAcc.Status.Status == status && userAcc.Status.Error == errMsg {
-		// Nothing changed
+// wrapErrorWithStatusUpdate wraps the error and update the user account status. If the update failed then log the error.
+func (r *ReconcileUserAccount) wrapErrorWithStatusUpdate(logger logr.Logger, userAcc *toolchainv1alpha1.UserAccount, statusUpdater func(userAcc *toolchainv1alpha1.UserAccount, message string) error, err error, format string, args ...interface{}) error {
```

Comment:
it looks like the number of parameters is growing - we could think about creating a new struct that would keep apart from the client also the logger and UserAccount so we don't have to pass it to all methods every time (it doesn't have to be part of this PR of course)